Wikipedia/UI-V5/WMFArticleContainerViewController.m

```diff
@@ -557,6 +564,34 @@ - (void)sectionEditorFinishedEditing:(SectionEditorViewController*)sectionEditor
     [self fetchArticle];
 }
 
+#pragma mark - UIViewControllerPreviewingDelegate
+
+- (void)configureLinkPreviewingDelegation {
+    id <UIViewControllerPreviewing>pc = [self registerForPreviewingWithDelegate:self sourceView:[self.webViewController.webView wmf_browserView]];
+    for (UIGestureRecognizer* r in [self.webViewController.webView wmf_browserView].gestureRecognizers) {
+        [r requireGestureRecognizerToFail:pc.previewingGestureRecognizerForFailureRelationship];
+    }
+}
+
+- (nullable UIViewController*)previewingContext:(id<UIViewControllerPreviewing>)previewingContext
+                      viewControllerForLocation:(CGPoint)location {
+    JSValue* element          = [self.webViewController htmlElementAtLocation:location];
+    MWKTitle* titleForElement = [self.webViewController titleForHTMLElement:element];
+    if (titleForElement) {
+        self.webViewController.isPeeking = YES;
+        previewingContext.sourceRect     = [self.webViewController rectForHTMLElement:element];
+        return [[WMFArticleContainerViewController alloc] initWithArticleTitle:titleForElement
+                                                                     dataStore:self.dataStore
+                                                               discoveryMethod:self.discoveryMethod];
+    }
+    return nil;
```

Comment:
What about external links? Should we just be able to get the url back and peek that as well?